### PR TITLE
fix(Scripts/Quest): Improved Varedis Must Be Stopped

### PR DESCRIPTION
**Pull request by grimgravy**
## Changes Proposed:
-  added the book and used only in Varedis
-  added remove metamorphosis

## SOURCE:

https://www.youtube.com/watch?v=D0GecU9-gUE (58:25 minutos de vídeo)

## How to Test the Changes:

1.  .quest add 10651 (Altar of Sha'tar)
2.  .quest add 10692 (Sanctum of the Stars)

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Spells/spell_quest.cpp [hunk 1]
```diff
@@ -2448,6 +2448,26 @@ class spell_q4735_collect_rookery_egg : public SpellScript
...
 };
 
+enum book
+{
+    SPELL_METAMORPHOSIS   = 36298
+};
+
+class spell_q10651_q10692_book_of_fel_names : public SpellScript
+{
+    PrepareSpellScript(spell_q10651_q10692_book_of_fel_names)
```
Comment (Nefertumm) [inline src/server/scripts/Spells/spell_quest.cpp:2458]: Missing semicolon

File: data/sql/updates/pending_db_world/rev_1689813586760998100.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+--
+DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_q10651_q10692_book_of_fel_names';
+INSERT INTO `spell_script_names` (`spell_id`,`ScriptName`) VALUES
+(36298, 'spell_q10651_q10692_book_of_fel_names');
+
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 17) AND (`SourceGroup` = 0) AND (`SourceEntry` = 37906) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 31) AND (`ConditionTarget` = 1) AND (`ConditionValue1` = 3) AND (`ConditionValue2` = 21178) AND (`ConditionValue3` = 0);
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(17, 0, 37906, 0, 0, 31, 1, 3, 21178, 0, 0, 0, 0, '', 'Rip the pages from this book when Varedis uses Metamorphosis to deprive him of his power.');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1689813586760998100.sql:8]: this comment is not descriptive of what the conditions does...

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1689813586760998100.sql:8]: I didn't understand about the comment. What is wrong with it?

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1689813586760998100.sql:8]: it does not describe the condition. It sounds more like a fairytale

Comment (grimgravy) [reply data/sql/updates/pending_db_world/rev_1689813586760998100.sql:8]: It is not a fairy tale. This is the description of the spell, it serves with intention to help understand about it. Although it doesn't sound like the condition, but it talks about the specific target. What is the purpose of it? Is it to draw attention or who is in doubt about targeting

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1689813586760998100.sql:8]: change it to
`Book of Fel Names can only target Varedis`

iirc that is what the condition actually does

Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1689813586760998100.sql:8]:
```suggestion
(17, 0, 37906, 0, 0, 31, 1, 3, 21178, 0, 0, 0, 0, '', 'Book of Fel Names can only target Varedis');
```

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**